src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -12,37 +12,66 @@
 namespace Symfony\Component\Serializer\Exception;
 
 /**
- * @deprecated since Symfony 6.3, use {@see MissingConstructorArgumentException} instead
- *
  * @author Maxime VEBER <maxime.veber@nekland.fr>
  */
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
-     * @var string[]
+     * @var string|null
+     */
+    private $class;
+
+    /**
+     * @var MissingConstructorArgumentException[]
      */
-    private $missingArguments;
+    private $missingConstructorArgumentExceptions = [];
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    /**
+     * @param MissingConstructorArgumentException[] $missingArguments
+     * @param ?class-string $class
+     */
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [], string $class = null)
     {
-        if (!$this instanceof MissingConstructorArgumentException) {
-            trigger_deprecation('symfony/serializer', '6.3', 'The "%s" class is deprecated, use "%s" instead.', __CLASS__, MissingConstructorArgumentException::class);
+        $this->missingConstructorArgumentExceptions = $missingArguments;
+        $classes = array_unique(array_map(fn (MissingConstructorArgumentException $missingConstructorArgumentException) => $missingConstructorArgumentException->getClass(), $missingArguments));
+        if (count($classes) > 1) {
+            throw new \InvalidArgumentException(sprintf(
+                'All instances of "%s" must concern the same class. Classes given : "%s".',
+                MissingConstructorArgumentException::class,
+                implode('", "', $classes)
+            ));
         }
-
-        $this->missingArguments = $missingArguments;
+        if ($class !== null && $class !== $classes[0]) {
+            throw new \InvalidArgumentException(sprintf(
+                'The class given in the constructor "%s" must be the same as the one given in the "%s" instances which is "%s".',
+                $class,
+                MissingConstructorArgumentException::class,
+                $classes[0]
+            ));
+        }
+        $this->class = $class ?: $classes[0];
```

Comment:
Same here
```suggestion
        $this->class = $class ?: ($classes[0] ?? null);
```

Reply:
This shall never be the case here IMO as the above if checks it and throws an exception.
Am I missing something ?

Reply:
If `$class` is null and `$classes` is `[]`, we won't enter the upper if, and won't throw any exception, right?